src/server/external-services/pathway-commons.js

```diff
@@ -17,109 +17,108 @@ const fetchOptions = {
-  const keywords = await _processPhrase(inputString);
-  const phrase = _sanitize(inputString);
-  // return three search query candidates: the first one is the fastest, the last - slowest
+// generate three search query candidates: the first one is the fastest, the last - slowest
+let generateSearchQueries = async inputString => {
+  let phrase = sanitize( inputString );
+  let entities = await extractEntityIds( inputString );
   return [
-    '(name:' + phrase + ') OR (' + 'name:*' + phrase + '*) OR (' + keywords.join(' AND ') + ')',
-    '(' + keywords.join(' OR ') + ')',
+    '(name:' + phrase + ') OR (' + 'name:*' + phrase + '*) OR (' + entities.join(' AND ') + ')',
+    '(' + entities.join(' OR ') + ')',
     inputString //"as is" (won't additionally escape Lucene query syntax, spaces, etc.)
   ];
 };
 
-//Pathway Commons HTTP GET request; options.cmd = 'pc2/get', 'pc2/search', 'pc2/traverse', 'pc2/graph', etc.
-const query = async (queryObj) => {
-  queryObj.user = 'app-ui';
-  let cmd = queryObj.cmd || 'pc2/get';
-  //TODO: (not critical) client app's sends useless parameters to the PC server: cmd, lt, gt
-  const url = config.PC_URL + cmd + '?' + qs.stringify(queryObj);
-  return fetch(url, fetchOptions)
-    .then(res => (cmd=='pc2/get'||cmd=='pc2/graph')?res.text():res.json())
-    .catch((e) => {
-      logger.error('query ' + queryObj + ' failed - ' + e);
-      return null;
-    });
-};
-
 // A fine-tuned PC search to improve relevance of full-text search and filter out unwanted hits.
 // The argument (query object) has the following fields:
 //  - q: user input - search query string
 //  - type: BioPAX type to match/filter by
 //  - lt: max graph size result returned
 //  - gt: min graph size result returned
-const _search = async (args) => {
-  const minSize = args.gt || 0;
-  const maxSize = args.lt || 250;
-  //analyse the input string, generate specific (lucene) search sub-queries
-  const queryString = args.q.trim();
-  const queries = await _processQueryString(queryString);
-  for (let q of queries) {
-    args.cmd = 'pc2/search'; //PC command
-    args.q = q; //override initial query.q string with the sub-query q
-    const searchResult = await query(args); //up to 100 hits at once; if we need more, then must use 'page' parameter...
-    const searchSuccess = searchResult != null;
-    if (searchSuccess && searchResult.searchHit.length > 0) {
-      const filteredResults = searchResult.searchHit.filter(hit => {
-        const size = hit.numParticipants ? hit.numParticipants : 0;
-        return minSize < size && size < maxSize;
-      });
-      if (filteredResults.length > 0) {
-        return filteredResults;
-      }
+let search = async opts => {
+  let { gt:minSize = 0, lt:maxSize = 250, q } = opts;
+  let queryStrings = await generateSearchQueries( q.trim() );
+
+  for( let queryString of queryStrings ) {
+    let queryOpts = _.assign( opts, { cmd: 'pc2/search', q: queryString } );
```

Comment:
I think @IgorRodchenkov mentioned dropping `gt` and `lt` from `queryOpts` since PC WS doesn't use them.